File: src/main/java/me/zeroeightsix/kami/module/modules/misc/NoEntityTrace.java
package me.zeroeightsix.kami.module.modules.misc;

import me.zeroeightsix.kami.module.ModulePlay;
import me.zeroeightsix.kami.setting.Setting;
import me.zeroeightsix.kami.setting.Settings;

/**
 * Created by 086 on 8/04/2018.
 */
@ModulePlay.Info(name = "NoEntityTrace", category = ModulePlay.Category.MISC, description = "Blocks entities from stopping you from mining")
public class NoEntityTrace extends ModulePlay {

    private Setting<TraceMode> mode = register(Settings.e("Mode", TraceMode.DYNAMIC));

    private static NoEntityTrace INSTANCE;

    public NoEntityTrace() {
        NoEntityTrace.INSTANCE = this;
    }

    public static boolean shouldBlock() {
        return INSTANCE.isEnabled() && (INSTANCE.mode.getValue() == TraceMode.STATIC || mc.interactionManager.isBreakingBlock());
    }

    private enum TraceMode {
        STATIC, DYNAMIC
    }
}

File: src/main/java/me/zeroeightsix/kami/feature/module/misc/NoEntityTrace.kt
package me.zeroeightsix.kami.feature.module.misc

import io.github.fablabsmc.fablabs.api.fiber.v1.annotation.Setting
import me.zero.alpine.listener.EventHandler
import me.zero.alpine.listener.EventHook
import me.zero.alpine.listener.Listener
import me.zeroeightsix.kami.event.events.TargetEntityEvent
import me.zeroeightsix.kami.feature.module.Module

/**
 * Created by 086 on 8/04/2018.
 */
@Module.Info(
    name = "NoEntityTrace",
    category = Module.Category.MISC,
    description = "Blocks entities from stopping you from mining"
)
object NoEntityTrace : Module() {
    @Setting
    var traceMode = TraceMode.DYNAMIC

    enum class TraceMode {
        STATIC, DYNAMIC
    }

    @EventHandler
    val targetListener = Listener(EventHook<TargetEntityEvent> {
        if (traceMode == TraceMode.STATIC || mc.interactionManager?.isBreakingBlock == true)
            it.trace = null
    })
}
